// Copyright (c) 2020, WAND Network Research Group
//                     Department of Computer Science
//                     University of Waikato
//                     Hamilton
//                     New Zealand
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software
// Foundation, Inc., 59 Temple Place, Suite 330,
// Boston,  MA 02111-1307  USA
//
// @Author : Brendon Jones (Original Disaggregated Router)
// @Author : Dimeji Fayomi

syntax = "proto3";

import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";
import "gobgp.proto"; 
import "attribute.proto";
import "capability.proto";
import "exabgp.proto";

service ControllerInterface {
  rpc SendPeerState (gobgpapi.ExaPeerState) returns (google.protobuf.Empty);
  rpc SendPeerOpen (gobgpapi.ExaPeerOpen) returns (google.protobuf.Empty);
  rpc SendPeerKeepalive (gobgpapi.ExaKeepalive) returns (google.protobuf.Empty);
  rpc SendUpdateEoR (gobgpapi.ExaUpdateEoR) returns (google.protobuf.Empty);
  rpc SendUpdate (gobgpapi.ExaUpdate) returns (google.protobuf.Empty);
  rpc GetPeerAddr (google.protobuf.Empty) returns (stream gobgpapi.GoBGPAddPeer);
}

service ExabgpInterface {
  rpc GetCtlrMsg (google.protobuf.Empty) returns (gobgpapi.ControllerToBGP);
}
